File: app/src/main/java/com/odd/ostrinov2/Ost.java
package com.odd.ostrinov2;

public class Ost {
    private String title, show, tags, url;
    private boolean isPlaying;

    private int id;

    Ost(){}

    Ost(String title, String show, String tags, String url) {
        this.title = title;
        this.show = show;
        this.tags = tags;
        this.url = url;
        isPlaying = false;
    }

    @Override
    public String toString() {
        return "Ost{title='" + title + '\'' +
                ", show='" + show + '\'' +
                ", tags='" + tags + '\'' +
                '}';
    }

    String getTags() {
        return tags;
    }

    void setTags(String tags) {
        this.tags = tags;
    }

    public String getShow() {
        return show;
    }

    public void setShow(String show) {
        this.show = show;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    String getUrl() {
        return url;
    }
    void setUrl(String url){
        this.url = url;
    }

    String getSearchString(){
        return title + ", " + show + ", " + tags;
    }

    public void setPlaying(boolean playing) {
        isPlaying = playing;
    }

    public boolean isPlaying(){
        return  isPlaying;
    }
}

File: app/src/main/java/com/odd/ostrinov2/Ost.kt
package com.odd.ostrinov2

import android.os.Parcel
import android.os.Parcelable
import com.odd.ostrinov2.tools.UtilMeths

class Ost(var title: String, var show: String, var tags: String, var videoId: String) : Parcelable {
    var id: Int = 0
    var url: String
        get() = UtilMeths.idToUrl(videoId)
        set(value) {
            videoId = UtilMeths.urlToId(value)
        }

    val searchString: String
        get() = "$title, $show, $tags"

    override fun toString(): String = "Ost{title='$title, show='$show, tags='$tags"

    override fun describeContents(): Int = 0

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(title)
        dest.writeString(show)
        dest.writeString(tags)
        dest.writeString(videoId)
        dest.writeInt(id)
    }

    constructor(parcel: Parcel) : this(
            parcel.readString(),
            parcel.readString(),
            parcel.readString(),
            parcel.readString()) {
        id = parcel.readInt()
    }

    companion object CREATOR : Parcelable.Creator<Ost> {
        override fun createFromParcel(parcel: Parcel): Ost = Ost(parcel)

        override fun newArray(size: Int): Array<Ost?> = arrayOfNulls(size)
    }
}
